### PR TITLE
[AIRFLOW-3150] Make execution_date templated in TriggerDagRunOperator

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3150

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
* `test_trigger_dagrun_with_str_execution_date`
* `test_trigger_dagrun_with_templated_execution_date`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/operators/dagrun_operator.py [hunk 1]
```diff
@@ -64,10 +66,26 @@ def __init__(
...
         self.python_callable = python_callable
         self.trigger_dag_id = trigger_dag_id
-        self.execution_date = execution_date
+
+        if isinstance(execution_date, datetime.datetime):
+            self.execution_date = execution_date.isoformat()
+        elif isinstance(execution_date, six.string_types):
+            self.execution_date = execution_date
+        elif execution_date is None:
+            self.execution_date = execution_date
```
Comment (feng-tao) [inline airflow/operators/dagrun_operator.py:75]: if execution_date is None, should we raise error ?

Comment (kaxil) [reply airflow/operators/dagrun_operator.py:75]: No, it is the default value as well and will also break backwards compatiblility.

Comment (kaxil) [reply airflow/operators/dagrun_operator.py:75]: It uses the current time when execution date is None

**Comment by kaxil** [comment]
@Fokko Any opinion?

**Comment by Fokko** [comment]
@kaxil Can you resolve the conflicts?

**Comment by kaxil** [comment]
@Fokko Done :)

**Comment by kaxil** [comment]
@Fokko Done :)

**Review by feng-tao — APPROVED** [review]
lgtm, thanks @kaxil

**Pull request merged**